branchpro/models.py

```diff
@@ -42,3 +43,106 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    """BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
```

Comment:
It would be great if you could addd a paragraph here that explains the model. You can maybe be inspired how Robin explained it on the exercise sheet. Ideally you also type up the formula. There is a reference in the issue to pints which shows how to do this.